internal/integration_test/integration_test.go

```diff
@@ -133,7 +134,7 @@ func TestTCPEcho(t *testing.T) {
 	const testTimeout = 200 * time.Millisecond
 	testMetrics := &service.NoOpTCPMetrics{}
 	authFunc := service.NewShadowsocksStreamAuthenticator(cipherList, &replayCache, testMetrics)
-	handler := service.NewTCPHandler(authFunc, testMetrics, testTimeout)
+	handler := service.NewTCPHandler(authFunc, testMetrics, testTimeout, 0)
```

Comment:
There's no way for you to specify the fwmark, which effectively makes it dead code.

Perhaps you have a fork and want the basic functionality here? As I mention below, let's inject the target dialers instead.

For the fwmark, that could be a config option, but it needs a bit of discussion, since the config is in flux. Adding @sbruens 

What is your specific need? Could you please clarify why you need the fwmark?

Reply:
fwmark can be used in conjunction with other Linux networking features like cgroups, network namespaces, ip route and TC (Traffic Control) for sophisticated network management:

- Flexible traffic management without altering packet contents
- Can be used with routing tables for advanced routing decisions
- Enables policy-based routing
- Load balancing across multiple internet connections
- Implementing quality of service (QoS) policies
- Marking packets for specific firewall rules